docs/content/advanced/backend/upgrade-guides/1-7-1.md

```diff
@@ -0,0 +1,113 @@
+
+## Actions Required
+
+### Run Migrations
+
+In the directory of your Medusa server, run the following command after updating the server:
+
+```bash
+medusa migrations run
+```
+
+### Run Migration Script
+
+Following the fix gift cards calculation, you also need to run a migration script after updating the server.
+
+Start by adding the following environment variables in `.env`:
+
+```bash
+TYPEORM_CONNECTION=postgres
+TYPEORM_URL=<DATABASE_URL>
+TYPEORM_USERNAME=<DATABASE_USERNAME>
+TYPEORM_PASSWORD=<DATABASE_PASSWORD>
+TYPEORM_DATABASE=<DATABASE_DATABASE>
+TYPEORM_ENTITIES=./node_modules/@medusajs/medusa/dist/models/*.js
+TYPEORM_MIGRATIONS=./node_modules/@medusajs/medusa/dist/migrations/*.js
+```
+
+Make sure to replace `<DATABASE_URL>`, `<DATABASE_USERNAME>`, `<DATABASE_PASSWORD>`, and `<DATABASE_DATABASE>` with your database connection details.
+
+Then, run the following command in the root directory of your Medusa server:
+
+```bash
+node ./node_modules/@medusajs/medusa/dist/scripts/gift-card-tax-rate-migration.js
+```
+
+### Change to JobSchedulerService
+
+In your loader file that creates a cron job, replace the use of `eventBus` to `jobSchedulerService`:
+
+```ts
+const myJob = async (container, options) => {
+  const jobSchedulerService = container.resolve("jobSchedulerService");
+  jobSchedulerService.create("my-job", {}, "0 0 * * *", async () => {
+    //...
+  })
+}
+```
+
+You can learn more in the “How to Create a Scheduled Job” documentation.
+
+### Change to Payment Provider
+
+This version of Medusa introduces a change in how payment providers are implemented. Mainly, the signature of the `createPayment` and `updatePayment` methods have changed, and the old signature is now deprecated.
+
+Although this change is currently backwards compatible, it is recommended to change the signature of these methods to the following:
+
+```ts
+import { PaymentContext, PaymentSessionResponse } from '@medusajs/medusa'
+
+async createPayment(context: PaymentContext): Promise<PaymentSessionResponse> {
+  //...
+}
+
+async updatePayment(paymentSessionData: PaymentSessionData, context: PaymentContext): Promise<PaymentSessionResponse> {
+  //...
+}
+```
+
+Where `context` in both `createPayment` and `updatePayment` is made up of the following properties:
+
+```ts
+type PaymentContext = {
+  cart: {
+    context: Record<string, unknown>
+    id: string
+    email: string
+    shipping_address: Address | null
+    shipping_methods: ShippingMethod[]
+  }
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
+}
+```
+
+So, you can pass the previous `cart` parameter inside the new `context` paramter.
+
+Furthermore, these methods are now expected to return `PaymentSessionResponse`. It is made up of the following properties:
+
+```ts
+type PaymentSessionResponse = {
+  update_requests: { customer_metadata: Record<string, unknown> }
```

Comment:
todo: The essence of this added property is that we've moved database mutations from payment plugins to the core to limit the capabilities and responsibilities of payment providers. 

You will use the `update_requests` property to tell the core engine to update a resource with its content. Right now, we only allow you to add metadata to our customer entity, but this will likely be extended in the future. 

This is exemplified in the Stripe payment plugin in which we create a Stripe customer for the provided email, and then send that Stripe customer id back to the core to be stored on the related Medusa customer metadata. 

Let me know if this makes sense!

Reply:
Ah got it! This is the question I had about for the payment provider guide as well. I'll update with these info